doc/custom-criteria.md

```diff
@@ -102,3 +102,17 @@ cfxr.set_options(custom_criteria=salt_criteria)
 
 ds.cf[["salinity"]]
 ```
+
+Here is an example of a more complicated custom criteria, which requires the package `regex` to be installed since a behavior (allowing global flags like "(?i)" for matching case insensitive) was recently deprecated in the `re` package. The custom criteria, called "vocab", matches – case insensitive – to the variable alias "sea_ice_u" a variable whose name includes "sea" and "ice" and "u" but not "qc" or "status", or "sea" and "ice" and "x" and "vel" but not "qc" or "status".
+
+```{code-cell}
```

Comment:
ideally this will fail and you'll need to add `regex` to `ci/docs.yml`

Let's also link to the `regex` documentation somewhere

Reply:
I think it ran because I *think* it is still a deprecation warning with Python 3.8 but becomes an error with Python 3.9 and the docs are on 3.8. But I did check the RTD build log and I don't see the deprecation warning in them. Still I'll go add `regex` to the docs env and link to the regex package docs.

Reply:
Ok I just added a commit with these items.